File: solutions/0739-daily-temperatures/solution.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
    const stack = [];
    const output = new Array(temperatures.length).fill(0);
    
    for(let i = 0; i < temperatures.length; i++){
        const temp = temperatures[i];
       
        while(stack.length > 0 && stack[stack.length-1][0] < temp){
            let prevDay = stack.pop()[1];        
            output[prevDay] = i - prevDay;
        }  
        
        stack.push([temp,i]);
    }
    
    return output;
};

File: solutions/0739-daily-temperatures/solution.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        out = [0] * len(temperatures)

        stack = []

        for i in range(len(temperatures)):
            t = temperatures[i]

            while stack and t > stack[-1][0]:
                prev_index = stack[-1][1]
                out[prev_index] = i - prev_index
                stack.pop()

            stack.append([t,i])

        return out 
    
